src/Compilers/CSharp/Portable/Lowering/DiagnosticsPass_ExpressionTrees.cs

```diff
@@ -755,6 +755,23 @@ public override BoundNode VisitUserDefinedConditionalLogicalOperator(BoundUserDe
                 {
                     Error(ErrorCode.ERR_ExpressionTreeContainsAbstractStaticMemberAccess, node);
                 }
+
+                if (binary.GetIsNewExtensionMember())
+                {
+                    // PROTOTYPE: Communicate to LDM
+                    // An expression tree factory isn't happy in this case. It throws
+                    //            System.ArgumentException : The user-defined operator method 'op_BitwiseOr' for operator 'OrElse' must have associated boolean True and False operators.
```

Comment:
Would it work if there were corresponding `op_True` and `op_False` operators also defined in the extension class?